### PR TITLE
[Misc] Enable YAML File for Llumnix Configuration

### PR DESCRIPTION
**Pull request by KuilongCui**
1. Enable YAML File for Llumnix Configuration
2. Update pylint rule for tests
3. Fix pylint error

### CODE REVIEW

File: llumnix/llumnix_config.py
Comment (zhypku) [inline llumnix/llumnix_config.py:1]: why not just config.py? `llumnix/llumnix_config.py` looks a bit duplicated

File: llumnix/config/config.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# limitations under the License.
+
+import os
+from typing import List
+import yaml
+from yacs.config import CfgNode
+
+BASE_KEY = "_BASE_"
+
+class LlumnixConfig(CfgNode):
```
Comment (zhypku) [inline llumnix/config/config.py:21]: We have `llumnix/llumnix_config.py`, `llumnix/config/config.py`, and even a `LlumnixConfig` class in `llumnix/config/config.py`. This looks confusing. Please clarify the naming of these directories / files / classes

File: docs/Quickstart.md [hunk 2]
```diff
@@ -62,24 +61,23 @@ During the execution of serving deployment, Llumnix will:
...
-                --port $PORT \
-                --initial-instances $INITIAL_INSTANCES \
-                --launch-ray-cluster \
+                --config-file $CONFIG_PATH \
                 --model $MODEL_PATH \
                 --engine-use-ray \
                 --worker-use-ray \
-                --max-model-len 4096
+                --max-model-len 4096 \
+                MANAGER.INITIAL_INSTANCES 1 \
```
Comment (zhypku) [inline docs/Quickstart.md:77]: why we two kinds of different parameter styles `--max-model-len` and `MANAGER.INITIAL_INSTANCES`? we should have the same style for CLI flags (it's ok to use another style in your config file, but do not mix two styles in CLI)

File: llumnix/arg_utils.py [hunk 3]
```diff
@@ -16,12 +16,11 @@
 import argparse
 from typing import Tuple
 
-from llumnix.config import GlobalSchedulerConfig, MigrationConfig
-
+from llumnix.llumnix_config import GlobalSchedulerConfig, MigrationConfig
```
Comment (zhypku) [inline llumnix/arg_utils.py:19]: To echo my previous comment on naming:
```
from llumnix.llumnix_config import GlobalSchedulerConfig, MigrationConfig
from llumnix.config.config import LlumnixConfig
```

This code is pretty confusing. Please refine these names

File: llumnix/entrypoints/vllm/api_server.py [hunk 4]
```diff
@@ -224,34 +225,25 @@ async def is_ready():
...
-    parser.add_argument('--disable-log-requests-server',
-                        action='store_true',
-                        help='disable logging requests in server')
-    parser.add_argument("--ray-cluster-port", type=int, default=30050)
-    parser.add_argument('--launch-ray-cluster',
-                        action='store_true',
-                        help='if launch ray cluster in api server')
-
-    parser = EngineManagerArgs.add_cli_args(parser)
+    parser.add_argument("--config-file", default="", metavar="FILE", help="path to config file")
```
Comment (zhypku) [inline llumnix/entrypoints/vllm/api_server.py:228]: can we allow to override the parameters using CLI? it's sometimes more convenient to use CLI, for example, when you want to change parameters using a shell script

**Comment by s5u13b** [comment]
Consider a better tag, [improve] sounds a bit odd.

File: docs/Quickstart.md (hunk 2 above)
Comment (KuilongCui) [reply docs/Quickstart.md:84]: In gereral, MANAGER.INITIAL_INSTANCES should not be set from cli. It should be set in config.yml. Here, just to give an example how to overwrite the parameter from cli.

File: llumnix/config/config.py (hunk 1 above)
Comment (KuilongCui) [reply llumnix/config/config.py]: yes, this is very confusing ~!

llumnix/llumnix_config.py contains internal config structures (MigrationConfig and GlobalSchedulerConfig). Rename llumnix/llumnix_config.py -> llumnix/internal_config.py.

llumnix/config folder contains files about llumnix config parameters. rename llumnix/config/config.py ->  llumnix/config/llumnix_config.py.  And to make it concise, import LlumnixConfig in llumnix config package (see llumnix/config/__init__.py). So, just need " from llumnix.config import LlumnixConfig", not "from llumnix.config.llumnix_config import LlumnixConfig"

File: docs/Quickstart.md (hunk 2 above)
Comment (KuilongCui) [reply docs/Quickstart.md:84]: when use config_file, we start the servive by :

python -m llumnix.entrypoints.vllm.api_server \
    --config-file $CONFIG_PATH \
    # vLLM arguments ...
    # Llumnix arguments ...
    ...

Llumnix arguments mean MANAGER.INITIAL_INSTANCES ... and so on. But in gereral, MANAGER.INITIAL_INSTANCES should not be set from cli. It should be set in config.yml. Here, just to give an example how to overwrite the parameter from cli.

File: llumnix/llumnix_config.py
Comment (KuilongCui) [reply llumnix/llumnix_config.py:1]: there exist llumnix/config folder.
 config.py is confusing when from llumnix.config import ...  
from the config config folder or config.py?

File: docs/Quickstart.md (hunk 2 above)
Comment (zhypku) [reply docs/Quickstart.md:84]: My point is, all the cli parameters should have the same naming style. We should not use `MANAGER.INITIAL_INSTANCES` in cli, instead, use `--initial-instances`. We then assign the value to the corresponding parameter in the internal config class.

And, if you don't want users to set a parameter from CLI, then just delete this parameter from CLI. If you include the parameter in CLI, then use the CLI naming style.

**Comment by KuilongCui** [comment]
Temporarily close this PR; the YAML config is not necessary right now.

**Pull request closed without merging**